templates/snmpd.conf.erb

```diff
@@ -14,64 +14,42 @@ agentaddress <%= @agentaddress.join(',') %>
 
 # ------------------------------------------------------------------------------
 # Traditional Access Control
-<% if @ro_community and (@ro_community.size > 0) -%>
-<% if (@ro_network.is_a?(Array)) and (@ro_network.count > 0) -%>
-<% @ro_network.each do |ro_net| -%>
-rocommunity <%= @ro_community %> <%= ro_net %>
-<% end -%>
-<% else -%>
-rocommunity <%= @ro_community %> <%= @ro_network %>
-<% end -%>
-<% end -%>
-<% if @ro_community6 and (@ro_community6.size > 0) -%>
-<% if (@ro_network6.is_a?(Array)) and (@ro_network6.count > 0) -%>
-<% @ro_network6.each do |ro_net6| -%>
-rocommunity6 <%= @ro_community6 %> <%= ro_net6 %>
-<% end -%>
-<% else -%>
-rocommunity6 <%= @ro_community6 %> <%= @ro_network6 %>
-<% end -%>
-<% end -%>
-<% if @rw_community and (@rw_community.size > 0) -%>
-<% if (@rw_network.is_a?(Array)) and (@rw_network.count > 0) -%>
-<% @rw_network.each do |rw_net| -%>
-rwcommunity <%= @rw_community %> <%= rw_net %>
-<% end -%>
-<% else -%>
-rwcommunity <%= @rw_community %> <%= @rw_network %>
-<% end -%>
-<% end -%>
-<% if @rw_community6 and (@rw_community6.size > 0) -%>
-<% if (@rw_network6.is_a?(Array)) and (@rw_network6.count > 0) -%>
-<% @rw_network6.each do |rw_net6| -%>
-rwcommunity6 <%= @rw_community6 %> <%= rw_net6 %>
-<% end -%>
-<% else -%>
-rwcommunity6 <%= @rw_community6 %> <%= @rw_network6 %>
-<% end -%>
-<% end -%>
+<%- [*@ro_community].compact.each do |c| -%>
```

Comment:
The [*@ro_community] syntax will take a variable and convert it into an array if it is not already. This allows us to really trim down this file.